tests/unit/spec/AdalSpec.js

```diff
@@ -807,6 +811,41 @@ describe('Adal', function () {
                 + '&client-request-id=44444444-4444-4444-8444-444444444444' + adal._addLibMetadata() + '&prompt=none&login_hint=test%40testuser.com&domain_hint=testuser.com', 'adalRenewFrametoken.resource1', 'token.resource1');
 
     });
+
+    it('checks the deserialize method for extracting idToken', function () {
+        var obj = adal._deserialize(VALID_URLFRAGMENT);
+        expect(obj.id_token).toBe(IDTOKEN_MOCK);
+        expect(obj.state).toBe(STATE);
+        expect(obj.session_state).toBe(SESSION_STATE);
+
+        obj = adal._deserialize(INVALID_URLFRAGMENT);
+        expect(obj.id_token).toBeUndefined;
+        expect(obj.state).toBe(STATE);
+        expect(obj.session_state).toBe(SESSION_STATE);
+        expect(obj['id_token' + IDTOKEN_MOCK]).toBeUndefined;
+
+        adal._deserialize = function (query) {
```

Comment:
There is an issue with this pattern. Any tests that we add below this test will use the incorrect "deserialize" method as it is being overridden here. You can save the old function and reassign it to adal._deserialize.